test/functional/nested-validation.spec.ts

```diff
@@ -138,4 +138,57 @@ describe("nested validation", function () {
 
     });
 
+    it("should not validate nulls in nested array", () => {
+
+        class MySubClass {
+            @MinLength(5)
+            name: string;
+        }
+
+        class MyClass {
+            @ValidateNested({ each: true })
+            mySubClasses: MySubClass[];
+        }
+
+        const model = new MyClass();
+        model.mySubClasses = [null, new MySubClass()];
+        model.mySubClasses[1].name = "my";
+
+        return validator.validate(model).then(errors => {
+            expect(errors[0].target).to.equal(model);
+            expect(errors[0].property).to.equal("mySubClasses");
+            expect(errors[0].children.length).to.equal(1);
+            expect(errors[0].children[0].children.length).to.equal(1);
+
+            const subError = errors[0].children[0].children[0];
+            subError.constraints.should.be.eql({minLength: "name must be longer than or equal to 5 characters"});
```

Comment:
Shouldn't we show a message regarding the null item instead of this?

Reply:
@amirsaber I don't think so. The original problem was related to unhandled exception in class-validator module. And this PR just fixes this issue (makes class-validator don't regard nulls as objects in nested array). It's just input data filtering.
According to https://github.com/typestack/class-validator#validating-nested-objects ValidateNested() will only validate nested object if it's present. But as for me, checking for array content should be performed in another validator, such as `@IsDefined({ each: true })`.
Imho the situation when you'll have an array that can contain either nulls or objects with specified format isn't impossible.

Reply:
We should add note about skipping non-object values in `@ValidateNested()` to the docs. Also small recommendation about using `@IsObject()` or `@IsDefined()` to handle this case.